CheckYourEligibility.API/Controllers/ApplicationController.cs

```diff
@@ -350,4 +353,46 @@ public async Task<ActionResult> DeleteApplication(string guid)
             return BadRequest(new ErrorResponse { Errors = [new Error { Title = ex.Message }] });
         }
     }
+
+    /// <summary>
+    /// Restores an archived application by GUID
+    /// </summary>
+    /// <param name="guid">The application GUID to restore</param>
+    [ProducesResponseType(typeof(ApplicationStatusRestoreResponse), (int)HttpStatusCode.OK)]
+    [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.NotFound)]
+    [Consumes("application/json", "application/vnd.api+json;version=1.0")]
+    [HttpPost("/application/restore/{guid}")]
```

Comment:
I believe the endpoint should be PATCH /application/{guid}/restore if possible (to follow pattern) - or maybe even better you can let the application status restoration be a side-effect of the [HttpPatch("/application/{guid}")].

The user of the API shouldn't care that we're having to restore the status, so they might as well set it to a transient Restore status. Or use your restore endpoint above, that works too.